src/containers/CompanyContainer.tsx

```diff
@@ -1,19 +1,166 @@
 
-import { CompanyAddForm } from '@components/company';
+import { CompanyAddForm, DomainStatus } from '@components/company';
+
+import GoogleIconImage from 'assets/google.png';
+import NetlifyIconImage from 'assets/netlify.png';
+
+import { useAddDNSRecord } from 'hooks/apiHooks/company/useAddDNSRecord';
+import { useAddDomainAlias } from 'hooks/apiHooks/company/useAddDomainAlias';
+import { useCreateCompany } from 'hooks/apiHooks/company/useCreateCompany';
+import { useToast } from 'hooks/useToast';
+
+import type { CompanyFormProps } from 'interfaces';
+
+const RETRY_LIMIT = 3;
 
 export const CompanyContainer = () => {
+    const createCompany = useCreateCompany();
+    const addDNSRecord = useAddDNSRecord();
+    const addDomainAlias = useAddDomainAlias();
     const [searchParams] = useSearchParams();
 
+    const { showError } = useToast();
+
+    const [formCompanyId, setFormCompanyId] = React.useState('');
+    const [showDNSStatus, setShowDNSStatus] = React.useState(false);
+    const [showDomainAliasStatus, setShowDomainAliasStatus] =
+        React.useState(false);
+    const [dnsRetryCount, setDnsRetryCount] = React.useState(0);
+    const [domainAliasRetryCount, setDomainAliasRetryCount] = React.useState(0);
+
+    const isCreationStatusVisible = React.useMemo(() => {
+        return (
+            createCompany.isSuccess && showDNSStatus && showDomainAliasStatus
+        );
+    }, [createCompany.isSuccess, showDNSStatus, showDomainAliasStatus]);
+
+    const isCreatingCompany = React.useMemo(() => {
+        return (
+            createCompany.isLoading ||
+            addDNSRecord.isLoading ||
+            addDomainAlias.isLoading
+        );
+    }, [
+        addDNSRecord.isLoading,
+        addDomainAlias.isLoading,
+        createCompany.isLoading
+    ]);
+
+    const createDNSRecord = (companyId: string) => {
+        addDNSRecord.mutate(
+            {
+                params: { companyId },
+                requestBody: { name: `${companyId}.hunar.ai` }
+            },
+            {
+                onSettled: () => {
+                    setShowDNSStatus(true);
+                    setDnsRetryCount(count => count + 1);
+                }
+            }
+        );
+    };
+
+    const createDomainAlias = (companyId: string) => {
+        addDomainAlias.mutate(
+            {
+                params: { companyId },
+                requestBody: { domainAlias: `${companyId}.hunar.ai` }
+            },
+            {
+                onSettled: () => {
+                    setShowDomainAliasStatus(true);
+                    setDomainAliasRetryCount(count => count + 1);
+                }
+            }
+        );
+    };
+
+    const onSubmitClick = (form: CompanyFormProps) => {
+        createCompany.mutate(
+            {
+                params: { companyId: form.companyId },
+                requestBody: form
+            },
+            {
+                onSuccess: () => {
```

Comment:
why are we doing all of this here? and not in companyAddForm?

Reply:
Moved this method to `CompanyAddForm`